File: 3438/3438.c
int findValidPairOfAdjacentDigitsI(struct TreeNode* root) {
    if (!root) return 0;
    int l = findValidPairOfAdjacentDigitsI(root->left);
    int r = findValidPairOfAdjacentDigitsI(root->right);
    return 1 + (l > r ? l : r);
}

File: 3438/3438.go
func findValidPairOfAdjacentDigitsI(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := findValidPairOfAdjacentDigitsI(root.Left)
	r := findValidPairOfAdjacentDigitsI(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}